commands/role.bee.inc

```diff
@@ -0,0 +1,387 @@
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-create manager' => bt("Add role 'manager'."),
+      ),
+    ),
+    'role-delete' => array(
+      'description' => bt('Delete a role.'),
+      'callback' => 'role_delete_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'role' => bt('Role to delete'),
+      ),
+      'aliases' => array('rdel'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-delete manager' => bt("Delete role 'manager'."),
+      ),
+    ),
+    'role-add-perm' => array(
+      'description' => bt('Grant specified permission(s) to a role.'),
+      'callback' => 'role_add_permission_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'permissions' => bt('Permissions'),
+        'role' => bt('Role'),
+      ),
+      'aliases' => array('rap'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-add-perm  \'post comments\' \'anonymous user\'' => bt("Allow anon users to post comments."),
+        'bee role-add-perm  "\'view own unpublished content\' , \'view any unpublished content\' , \'view revisions\'" \'anonymous user\'' => bt("Grant multiple permissions to the anon users"),
+      ),
+    ),
+    'role-remove-perm' => array(
+      'description' => bt('Remove specified permission(s) from a role.'),
+      'callback' => 'role_remove_permission_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'permissions' => bt('Permissions'),
+        'role' => bt('Role'),
+      ),
+      'aliases' => array('rrp'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-remove-perm \'access content\' \'anonymous user\'' => bt("Hide content from anon users."),
+        'bee role-remove-perm  "\'view own unpublished content\' , \'view any unpublished content\' , \'view revisions\'" \'anonymous user\'' => bt("Remove multiple permissions from the anon users"),
+      ),
+    ),
+  );
+}
+
+/**
+ * Command callback: List all roles.
+ */
+function roles_bee_callback($arguments, $options) {
+  $rows = array();
+  $roles = user_roles(TRUE);
+  unset($roles['authenticated']);
+  if (!empty($options['role'])) {
+    // If a role has been specified, check if exists, and return error if
+    // not.
+    $role = strtolower($options['role']);
+    if (!array_key_exists($role, $roles)) {
+      bee_message(bt("The role '!role' doesn't exists!", array(
+        '!role' => $role,
+      )), 'error');
+      return;
+    }
+    $role_array = array(strtolower($options['role']));
+    $permissions = implode('\',\'', user_role_permissions($role_array));
+    if (!empty($permissions)) {
+      $permissions = '\''.$permissions.'\'';
+    }
+    bee_message(bt("The !role role has the following permissions granted: !permissions", array(
+      '!role' => $options['role'],
+      '!permissions' => $permissions,
+    )));
```

Comment:
1. The output should be inside the positive pathway
2. Like the `permissions_bee_callback()` this needs to be output as text, not as a message.
3. There needs to be a negative pathway (i.e. if no permissions have been granted). It's not necessarily an error but the message needs to be different.  Perhaps an info message? The role will have all `authenticated` permissions, so it's not like they won't have permissions, so the message needs to reflect that.

Reply:
@nico8948 - still to do